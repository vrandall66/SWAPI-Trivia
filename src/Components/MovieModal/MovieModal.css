.MovieModalContainer {
  display: flex;
  flex-direction: column;
}

.MovieModal__btn--close {
  height: 42px;
  width: 42px;
}