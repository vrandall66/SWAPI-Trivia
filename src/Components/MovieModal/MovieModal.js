import React from "react";
import { Link } from "react-router-dom";
import "./MovieModal.css";
import lightsaberClose from "../../images/LightsaberCross.png";

const MovieModal = ({ currentMovie, updatePlanetModalState }) => {
  const { title, id, releaseDate, url } = currentMovie;
  return (
    <div className="MovieModalContainer">
      <div className='MovieModalContainer__div'>
        <input
          type="image"
          src={lightsaberClose}
          alt="Crossed lightsabers - close"
          className="MovieModal__btn--close"
          onClick={updatePlanetModalState}
        />
      </div>
      <h1 className="MovieModal__h1">{title}</h1>
      <h3 className="MovieModal__h3">Episode: {id}</h3>
      <h5 className="MovieModal__h5">Release Date:<br/> {releaseDate}</h5>
      <Link to={`/movies/${id}`}>
        <button
          type="button"
          className="MovieModal__btn--characters"
          onClick={updatePlanetModalState}
        >
          View Characters
        </button>
      </Link>
    </div>
  );
};

export default MovieModal;
