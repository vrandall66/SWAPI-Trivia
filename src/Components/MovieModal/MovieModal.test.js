import React from "react";
import { shallow } from "enzyme";
import MovieModal from "./MovieModal";

describe("MovieModal", () => {
  let wrapper;
  let currentMovie = {
    id: 1,
    releaseDate: "",
    title: "",
    url: ""
  };
  let updatePlanetModalState = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <MovieModal
        currentMovie={currentMovie}
        updatePlanetModalState={updatePlanetModalState}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
