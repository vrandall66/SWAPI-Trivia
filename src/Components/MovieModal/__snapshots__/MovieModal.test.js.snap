// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieModal should match the snapshot 1`] = `
<div
  className="MovieModalContainer"
>
  <input
    alt="Crossed lightsabers - close"
    className="MovieModal__btn--close"
    onClick={[MockFunction]}
    src="LightsaberCross.png"
    type="image"
  />
  <h1
    className="MovieModal__h1"
  />
  <h3
    className="MovieModal__h3"
  >
    1
  </h3>
  <h5
    className="MovieModal__h5"
  />
  <Link
    to="/movies/1"
  >
    <button
      className="MovieModal__btn--characters"
      onClick={[MockFunction]}
      type="button"
    >
      View Characters
    </button>
  </Link>
</div>
`;
